examples/platforms/nrf52840/gpio.c

```diff
@@ -0,0 +1,212 @@
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform GPIO driver on nrf52840.
+ *
+ */
+#include <openthread/config.h>
+#undef PACKAGE
+
+#include <stddef.h>
+#include <stdint.h>
+#include <inttypes.h>
+
+#include <common/logging.hpp>
+#include <openthread/types.h>
+#include <openthread/platform/toolchain.h>
+#include <openthread/platform/gpio.h>
+#include <openthread/platform/alarm-micro.h>
+#include <openthread/platform/alarm-milli.h>
+#include <openthread/platform/logging.h>
+#include <utils/code_utils.h>
+
+#include <drivers/clock/nrf_drv_clock.h>
+#include <hal/nrf_uart.h>
+#include <hal/nrf_gpio.h>
+#include <hal/nrf_gpiote.h>
+#include "platform-nrf5.h"
+
+#define IN_PIN 3
+
+#define TE_IDX_TO_EVENT_ADDR(idx)    (nrf_gpiote_events_t)((uint32_t)NRF_GPIOTE_EVENTS_IN_0 + \
+                                     (sizeof(uint32_t) * (idx)))
+
+static uint8_t sPinNumbers[30];
+otInstance *sInstance;
+static int state = 0;
+
+static uint32_t sPin = 3;
+
+void nrf5GpioInit(void)
+{
+    for (uint32_t i = 0; i < 30; i++)
+    {
+        nrf_gpio_pin_clear(i);
+    }
+}
+
+void otGpioInit(void)
+{
+    for (uint32_t i = 0; i < 30; i++)
+    {
+        nrf_gpio_pin_clear(i);
+    }
+}
+
+void nrf5GpioProcess(otInstance *aInstance)
+{
+    (void) aInstance;
+    if (nrf_gpio_pin_read(3) > 0)
+    {
+        if (state != 1)
+        {
+            otLogCritPlat(sInstance, " %d", state);
```

Comment:
use parameter `aInstance` and may be not necessary to keep the variable `sInstance`?